inc/Engine/Media/AboveTheFold/WarmUp/Controller.php

```diff
@@ -0,0 +1,96 @@
+<?php
+declare(strict_types=1);
+
+namespace WP_Rocket\Engine\Media\AboveTheFold\WarmUp;
+
+use WP_Rocket\Engine\Optimization\RegexTrait;
+use WP_Rocket\Engine\Common\JobManager\Managers\ManagerInterface;
+
+class Controller {
+	use RegexTrait;
+
+	/**
+	 * Above the fold Job Manager.
+	 *
+	 * @var ManagerInterface
+	 */
+	private $manager;
+
+	/**
+	 * Constructor
+	 *
+	 * @param ManagerInterface $manager Above the fold Job Manager.
+	 */
+	public function __construct( ManagerInterface $manager ) {
+		$this->manager = $manager;
+	}
+
+	/**
+	 * Process links fetched from homepage.
+	 *
+	 * @return void
+	 */
+	public function process_links(): void {
+		if ( ! $this->manager->is_allowed() ) {
+			return;
+		}
+
+		error_log( print_r( $this->fetch_links(), true ) );  // To be replaced by sending links to SaaS.
+	}
+
+	/**
+	 * Fetch links from homepage.
+	 *
+	 * @return array
+	 */
+	private function fetch_links(): array {
+		$home_url = home_url();
+		$args     = [
+			'user-agent' => 'WP Rocket/Pre-fetch Home Links',
+			'timeout'    => 60,
+		];
+
+		$response = wp_remote_get( $home_url, $args );
+
+		if ( ! is_array( $response ) && is_wp_error( $response ) ) {
+			return [];
+		}
```

Comment:
This is already done in `wp_remote_retrieve_body()`.

One thing to check though would be the response code, to make sure we receive a 200 response, with `wp_remote_retrieve_response_code()`.